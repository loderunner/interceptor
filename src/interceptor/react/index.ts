export { Provider as InterceptProvider, useIntercepts } from './intercept'
