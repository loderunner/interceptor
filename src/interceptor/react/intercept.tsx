import * as React from 'react'
import { createContext, useContext, useState } from 'react'

import * as intercept from '../intercept'

interface InterceptContextType {
  intercepts: intercept.Intercept[]
  addIntercept: (pattern: string) => void
  removeIntercept: (inter: intercept.Intercept) => void
}

const InterceptContext = createContext<InterceptContextType>({
  intercepts: [],
  addIntercept: () => {},
  removeIntercept: () => {},
})

interface Props {
  children: React.ReactNode
}

export const Provider = ({ children }: Props) => {
  const [intercepts, setIntercepts] = useState([...intercept.intercepts])

  const addIntercept = (pattern: string) => {
    intercept.addIntercept({ pattern, enabled: true })
    setIntercepts([...intercept.intercepts])
  }
  const removeIntercept = (inter: intercept.Intercept) => {
    intercept.removeIntercept(inter)
    setIntercepts([...intercept.intercepts])
  }

  return (
    <InterceptContext.Provider
      value={{ intercepts, addIntercept, removeIntercept }}
    >
      {children}
    </InterceptContext.Provider>
  )
}

export const useIntercepts = () => useContext(InterceptContext)
