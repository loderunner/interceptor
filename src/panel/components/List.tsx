import * as React from 'react'

interface Props {
  className?: string
  items?: React.ReactNode[]
  header?: React.ReactNode
}

const List = ({ className, header = '', items = [] }: Props) => {
  if (typeof header === 'string') {
    header = <div className="font-bold bg-slate-100">{header}</div>
  }

  return (
    <div className={`flex flex-col ${className}`}>
      {header}
      <div className="overflow-y-auto">{items}</div>
    </div>
  )
}

export default List
