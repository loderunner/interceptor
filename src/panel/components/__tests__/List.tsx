/* eslint-disable @typescript-eslint/ban-ts-comment */
import { cleanup, render } from '@testing-library/react'
import React from 'react'
import '@testing-library/jest-dom'

import List from '../List'

describe('[List]', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  afterEach(() => {
    cleanup()
  })

  it('should match empty snapshot', () => {
    const { container } = render(<List />)
    expect(container).toMatchSnapshot()
  })

  it('should match string header snapshot', () => {
    const header = 'Header title'
    const { container } = render(<List header={header} />)
    expect(container).toMatchSnapshot()
  })

  it('should match component header snapshot', () => {
    const header = <div className="header">Header Title</div>
    const { container } = render(<List header={header} />)
    expect(container).toMatchSnapshot()
  })

  it('should match header + string items snapshot', () => {
    const header = 'Header title'
    const items = ['Hello', 'World']
    const { container } = render(<List header={header} items={items} />)
    expect(container).toMatchSnapshot()
  })

  it('should match header + component items snapshot', () => {
    const header = 'Header title'
    const items = [
      <img key="img" src="//image.png"></img>,
      <label key="label">
        <input type="checkbox"></input>Hello World
      </label>,
    ]
    const { container } = render(<List header={header} items={items} />)
    expect(container).toMatchSnapshot()
  })
})
