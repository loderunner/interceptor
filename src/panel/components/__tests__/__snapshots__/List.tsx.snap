// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[List] should match component header snapshot 1`] = `
<div>
  <div
    class="flex flex-col "
  >
    <div
      class="header"
    >
      Header Title
    </div>
    <div
      class="flex flex-col overflow-y-auto"
    />
  </div>
</div>
`;

exports[`[List] should match empty snapshot 1`] = `
<div>
  <div
    class="flex flex-col "
  >
    <div
      class="font-bold bg-slate-100"
    />
    <div
      class="flex flex-col overflow-y-auto"
    />
  </div>
</div>
`;

exports[`[List] should match header + component items snapshot 1`] = `
<div>
  <div
    class="flex flex-col "
  >
    <div
      class="font-bold bg-slate-100"
    >
      Header title
    </div>
    <div
      class="flex flex-col overflow-y-auto"
    >
      <img
        src="//image.png"
      />
      <label>
        <input
          type="checkbox"
        />
        Hello World
      </label>
    </div>
  </div>
</div>
`;

exports[`[List] should match header + string items snapshot 1`] = `
<div>
  <div
    class="flex flex-col "
  >
    <div
      class="font-bold bg-slate-100"
    >
      Header title
    </div>
    <div
      class="flex flex-col overflow-y-auto"
    >
      Hello
      World
    </div>
  </div>
</div>
`;

exports[`[List] should match string header snapshot 1`] = `
<div>
  <div
    class="flex flex-col "
  >
    <div
      class="font-bold bg-slate-100"
    >
      Header title
    </div>
    <div
      class="flex flex-col overflow-y-auto"
    />
  </div>
</div>
`;
